test/index.test.js

```diff
@@ -184,31 +207,32 @@ test('Should record a reasonably accurate span duration with custom recorder', t
   const httpReporterUrl = 'http://0.0.0.0:9441/api/v2/spans'
   const PAUSE_TIME_MILLIS = 100
 
-  ctxImpl.scoped(() => {
-    fastify.register(zipkinPlugin, { recorder, serviceName, httpReporterUrl })
+  await new Promise((resolve) => {
+    ctxImpl.scoped(async () => {
+      await fastify.register(zipkinPlugin, { recorder, serviceName, httpReporterUrl })
 
-    fastify.get('/', (req, reply) => {
-      setTimeout(() => {
+      fastify.get('/', async (req, reply) => {
+        await sleep(PAUSE_TIME_MILLIS)
         reply.send({ hello: 'world' })
-      }, PAUSE_TIME_MILLIS)
+      })
+
+      fastify.inject(
```

Comment:
Since we added the `async`, I would write `const res = await fastify.inject()`
I would avoid mixin styles

Reply:
I missed it :D